zipline/algorithm.py

```diff
@@ -1212,64 +1107,19 @@ def cancel_order(self, order_param):
         self.blotter.cancel(order_id)
 
     @api_method
-    def add_history(self, bar_count, frequency, field, ffill=True):
-        data_frequency = self.sim_params.data_frequency
-        history_spec = HistorySpec(bar_count, frequency, field, ffill,
-                                   data_frequency=data_frequency,
-                                   env=self.trading_environment)
-        self.history_specs[history_spec.key_str] = history_spec
-        if self.initialized:
-            if self.history_container:
-                self.history_container.ensure_spec(
-                    history_spec, self.datetime, self._most_recent_data,
-                )
-            else:
-                self.history_container = self.history_container_class(
-                    self.history_specs,
-                    self.current_universe(),
-                    self.sim_params.first_open,
-                    self.sim_params.data_frequency,
-                    env=self.trading_environment,
-                )
-
-    def get_history_spec(self, bar_count, frequency, field, ffill):
-        spec_key = HistorySpec.spec_key(bar_count, frequency, field, ffill)
-        if spec_key not in self.history_specs:
-            data_freq = self.sim_params.data_frequency
-            spec = HistorySpec(
-                bar_count,
-                frequency,
-                field,
-                ffill,
-                data_frequency=data_freq,
-                env=self.trading_environment,
-            )
-            self.history_specs[spec_key] = spec
-            if not self.history_container:
-                self.history_container = self.history_container_class(
-                    self.history_specs,
-                    self.current_universe(),
-                    self.datetime,
-                    self.sim_params.data_frequency,
-                    bar_data=self._most_recent_data,
-                    env=self.trading_environment,
-                )
-            self.history_container.ensure_spec(
-                spec, self.datetime, self._most_recent_data,
-            )
-        return self.history_specs[spec_key]
-
-    @api_method
     @require_initialized(HistoryInInitialize())
-    def history(self, bar_count, frequency, field, ffill=True):
-        history_spec = self.get_history_spec(
+    def history(self, sids, bar_count, frequency, field, ffill=True):
+        if self.data_portal is None:
+            raise Exception("no data portal!")
+
+        return self.data_portal.get_history_window(
+            sids,
+            self.get_datetime(),
             bar_count,
             frequency,
             field,
-            ffill,
+            ffill
```

Comment:
Totally a style nit, but let me make the case: I notice a number of trailing commas were removed in this diff.  This adds a little noise, since they're not meaningful changes. But I've also found that this adds noise to diffs in future PRs, since adding a new argument will require changing this line back.  Since the trailing comma actually already works in both cases, I suggest leaving it and reducing the noise for everyone reading these PRs.